ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/EthereumWireProtocolConfiguration.java

```diff
@@ -68,15 +70,23 @@ public int getMaxGetNodeData() {
   }
 
   @Override
-  public boolean equals(final Object obj) {
-    if (!(obj instanceof EthereumWireProtocolConfiguration)) {
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
       return false;
     }
-    EthereumWireProtocolConfiguration other = ((EthereumWireProtocolConfiguration) obj);
-    return maxGetBlockHeaders == other.maxGetBlockHeaders
-        && maxGetBlockBodies == other.maxGetBlockBodies
-        && maxGetReceipts == other.maxGetReceipts
-        && maxGetNodeData == other.maxGetNodeData;
+    final EthereumWireProtocolConfiguration that = (EthereumWireProtocolConfiguration) o;
+    return maxGetBlockHeaders == that.maxGetBlockHeaders &&
+        maxGetBlockBodies == that.maxGetBlockBodies &&
+        maxGetReceipts == that.maxGetReceipts &&
+        maxGetNodeData == that.maxGetNodeData;
+  }
+
+  @Override
+  public int hashCode() {
```

Comment:
Optional. Maybe we could test equals and hashCode. Although it is non blocking.

Reply:
Equals is only used by tests and hash code isn't used (but should exist to match equals just for good measure).  I really only updated them to match our code style which is the generated code.